src/olympia/users/admin.py

```diff
@@ -227,39 +229,65 @@ def construct_search(field_name):
                 return "%s__iexact" % field_name[1:]
             elif field_name.startswith('@'):
                 return "%s__icontains" % field_name[1:]
-            else:
-                return "%s__icontains" % field_name
+            # Use field_name if it includes a lookup.
+            opts = queryset.model._meta
+            lookup_fields = field_name.split(models.constants.LOOKUP_SEP)
+            # Go through the fields, following all relations.
+            prev_field = None
+            for path_part in lookup_fields:
+                if path_part == 'pk':
+                    path_part = opts.pk.name
+                try:
+                    field = opts.get_field(path_part)
+                except FieldDoesNotExist:
+                    # Use valid query lookups.
+                    if prev_field and prev_field.get_lookup(path_part):
+                        return field_name
+                else:
+                    prev_field = field
+                    if hasattr(field, 'get_path_info'):
+                        # Update opts to follow the relation.
+                        opts = field.get_path_info()[-1].to_opts
```

Comment:
Where is all from above coming from? I somehow think it needs a little explanation what functionality and why it's adding that.

From what I get, it's just trying to resolve fields on relationships?

Reply:
I updated it from the django source code (the method comments note it's copied accross and then modified to support comma separated values)  - it changed in 2.2.  Although I didn't end up requiring any of the new functionality in the end so I could remove it again.

Reply:
Just add a comment where it's coming from and that should be fine.

Reply:
The comment is already there: https://github.com/mozilla/addons-server/pull/11388/files/b4f5a599ca21bfd51d74f6e5ae81c07da1d51764#diff-a45d02744bf3452dce7a83f87f8204a2L216

Reply:
Consider me blind, nevermind then!